silverback/application.py

```diff
@@ -22,6 +23,24 @@ class TaskData:
     handler: AsyncTaskiqDecoratedTask
 
 
+class SharedState(defaultdict):
+    def __init__(self):
+        # Any unknown key returns None
+        super().__init__(lambda: None)
+
+    def __getattr__(self, attr):
+        try:
+            super().__getattr__(attr)
+        except AttributeError:
+            return super().__getitem__(attr)
+
+    def __setattr__(self, attr, val):
+        try:
+            super().__setattr__(attr, val)
+        except AttributeError:
+            super().__setitem__(attr, val)
```

Comment:
question: What's up with this pattern?  I'm not familiar with the setattr/setitem differences and why we'd want to attempt both.  

Also, if this is just a `defaultdict`, why implement these at all?

Reply:
> question: What's up with this pattern? I'm not familiar with the setattr/setitem differences and why we'd want to attempt both.

`__setattr__` is `obj.attr = val`, `__setitem__` is `obj[attr] = val`

Usually, attempt `__setattr__` first in case there's some baseclass method you want to ensure is properly replicating

> Also, if this is just a `defaultdict`, why implement these at all?

I think you're right, I might be overoptimizing here